pkg/dns/dns.go

```diff
@@ -330,6 +328,32 @@ func (kd *KubeDNS) handleEndpointAdd(obj interface{}) {
 	}
 }
 
+func (kd *KubeDNS) handleEndpointUpdate(oldObj, newObj interface{}) {
+	kd.handleEndpointDelete(oldObj)
```

Comment:
This gives us a brief window during which resolution will fail.  We can't do that.  This seems to be a logically distinct change, please do a different PR if you want to tackle this.